core/App/screens/NameCreate.tsx

```diff
@@ -0,0 +1,107 @@
+import { useNavigation } from '@react-navigation/core'
+import { StackNavigationProp } from '@react-navigation/stack'
+import React, { useContext, useEffect, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { StyleSheet, Text, View } from 'react-native'
+import { SafeAreaView } from 'react-native-safe-area-context'
+import Icon from 'react-native-vector-icons/MaterialCommunityIcons'
+
+import Button, { ButtonType } from '../components/buttons/Button'
+import TextInput from '../components/inputs/TextInput'
+import { DispatchAction } from '../contexts/reducers/store'
+import { StoreContext } from '../contexts/store'
+import { useTheme } from '../contexts/theme'
+import { AuthenticateStackParams, Screens } from '../types/navigators'
+
+const NameCreate: React.FC = () => {
+  const { t } = useTranslation()
+  const [state, dispatch] = useContext(StoreContext)
+  const { ColorPallet } = useTheme()
+  const [firstName, onChangeFirstName] = useState('')
+  const [lastName, onChangeLastName] = useState('')
```

Comment:
This should likely follow hooks convention and be setFirstName & setLastName

Reply:
Weirdly this is the convention outlined in the TextInput documentation for React Native. Happy to change it though https://reactnative.dev/docs/textinput#props